coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala

```diff
@@ -241,6 +241,39 @@ import filodb.query.exec._
     PlanResult(Seq(stitchedPlan))
   }
 
+  /**
+   * Materializes a raw series plan and routes the query to either the Raw or DownSample cluster based on lookback time.
+   * - If the target time range falls within the raw data retention period, the query is routed to the Raw cluster.
+   * - Otherwise, it is routed to the DownSample cluster, In this case, it will have no data after
+   *   latestDownsampleTimestampFn because they are unavailable.
+   *
+   * Limitations:
+   * - Stitching data from the Raw and DownSample clusters is not supported due to schema differences.
+   *
+   * @param logicalPlan  The raw series logical plan to materialize.
+   * @param queryContext The QueryContext object.
+   * @return
+   */
+  private def materializeRawSeries(queryContext: QueryContext, logicalPlan: RawSeries): PlanResult = {
+    val timeRange = getTimeFromLogicalPlan(logicalPlan)
+
+    // use rawClusterMaterialize to handle range Raw data queries with range expression []
+    if(timeRange.startMs != timeRange.endMs){
+      logger.info("Materializing ranged raw series export against raw cluster:: {}", queryContext.origQueryParams)
+      return rawClusterMaterialize(queryContext, logicalPlan)
+    }
+    val lookbackMs = logicalPlan.lookbackMs.getOrElse(0L)
+    val offsetMs = logicalPlan.offsetMs.getOrElse(0L)
+    val (startTime, endTime) = (timeRange.endMs - lookbackMs - offsetMs, timeRange.endMs - offsetMs)
+    val execPlan = if (startTime > earliestRawTimestampFn) {
```

Comment:
I'll type out loud as a sanity-check: in the current model, all `RawSeries` are routed exclusively to the raw cluster. That means data will be missing when that `RawSeries` spanned across the raw/downsample boundary. The max duration of missing data is unbounded, but the data that is returned will be non-downsampled.

In this model, `RawSeries` are routed exclusively to the downsampled cluster when any downsampled data is required. When that happens, the duration of missing data is upper-bounded by the max duration until data is downsampled; additionally, the data returned is downsampled.

So, we're trading data granularity to reduce the (possibly significant) duration of missing data. Sounds like a win to me 👍